### PR TITLE
Revolutionaries can speak french now

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Adds French, which revolutionaries are able to speak and understand. Mimes can understand it as well, but to speak it they must break their oath.

image

# Wiki Documentation

Might be worth mentioning here:
https://wiki.yogstation.net/wiki/Revolution
https://wiki.yogstation.net/wiki/Mime

# Changelog

:cl:  
rscadd: Adds french
/:cl:

### CODE REVIEW

**Comment by TheBombMakerJL** [comment]
Speedmerge this please

**Comment by Therandomhoboo** [comment]
Gonna be a massive nerd, there is a drink named Cuba Libre which heals revvies which is basically a Cuban Drink which is kinda like spanish. So wouldn't that make them Spanish?

File: code/modules/antagonists/revolution/revolution.dm [hunk 1]
```diff
@@ -32,10 +32,12 @@
 
 /datum/antagonist/rev/apply_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	M.grant_language(/datum/language/french, source = LANGUAGE_REVOLUTIONARY)
 	update_rev_icons_added(M)
 
 /datum/antagonist/rev/remove_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	M.remove_language(/datum/language/french, source = LANGUAGE_REVOLUTIONARY)
```
Comment (ynot01) [inline code/modules/antagonists/revolution/revolution.dm]: You use `source =` here

File: code/modules/jobs/job_types/mime.dm [hunk 2]
```diff
@@ -65,6 +65,7 @@
 	if(visualsOnly)
 		return
 
+	H.grant_language(/datum/language/french, TRUE, TRUE, LANGUAGE_MIME)
```
Comment (ynot01) [inline code/modules/jobs/job_types/mime.dm:68]: but use `TRUE, TRUE` here?

**Comment by MajManatee** [comment]
now THIS is a good idea

**Comment by iamthesilly** [comment]
> Gonna be a massive nerd, there is a drink named Cuba Libre which heals revvies which is basically a Cuban Drink which is kinda like spanish. So wouldn't that make them Spanish?

france invented the guillotine and did more revolutions than cuba

File: code/modules/antagonists/revolution/revolution.dm [hunk 3]
```diff
@@ -32,10 +32,12 @@
 
 /datum/antagonist/rev/apply_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	M.grant_language(/datum/language/french, TRUE, TRUE, LANGUAGE_REVOLUTIONARY)
```
Comment (ynot01) [inline code/modules/antagonists/revolution/revolution.dm:35]: image

File: code/modules/antagonists/revolution/revolution.dm [hunk 4]
```diff
@@ -32,10 +32,12 @@
...
 	update_rev_icons_added(M)
 
 /datum/antagonist/rev/remove_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	M.remove_language(/datum/language/french, source = LANGUAGE_REVOLUTIONARY)
```
Comment (ynot01) [inline code/modules/antagonists/revolution/revolution.dm]: image

**Comment by Therandomhoboo** [comment]
> > Gonna be a massive nerd, there is a drink named Cuba Libre which heals revvies which is basically a Cuban Drink which is kinda like spanish. So wouldn't that make them Spanish?
> 
> france invented the guillotine and did more revolutions than cuba

Yeah but Che Guevara is basically the head figure for "Revolutions" I would argue. No one really knows of any famous french revolutionists that is widely known apart from Joan of Arc

**Review by ynot01 — APPROVED** [review]
Code looks fine, good luck with that icon, also kind of confusing that the beret that the mime starts with gives you a "french accent" and now this is also a thing

**Comment by Therandomhoboo** [comment]
> Code looks fine, good luck with that icon, also kind of confusing that the beret that the mime starts with gives you a "french accent" and now this is also a thing

As well as Artist gives you a french accent too

**Comment by JamieD1** [comment]
lol

**Comment by iamthesilly** [comment]
i was going to make the french beret let you speak french but in the middle of adding that i realized sec would just steal it from the mime or artist during revolutions in order to understand the revolutionaries and decided against it

**Comment by Marmio64** [comment]
even if almost nobody is going to use this language i'd probably still add more 'syllables' since 10 is a really small amount when they're all words.

**Comment by cuackles** [comment]
I'm not gonna be at my PC until tomorrow so could I get a photo of the sprite in full, as in a screenshot of the DMI or a PNG of it

**Comment by iamthesilly** [comment]
> I'm not gonna be at my PC until tomorrow so could I get a photo of the sprite in full, as in a screenshot of the DMI or a PNG of it

image

**Comment by cuackles** [comment]
> > I'm not gonna be at my PC until tomorrow so could I get a photo of the sprite in full, as in a screenshot of the DMI or a PNG of it
> 
> image
> 

Did you edit it ? It doesn't look like the one on the screenshot

**Comment by cuackles** [comment]
Looks a lot more clean and less fuzzy

**Comment by iamthesilly** [comment]
> Did you edit it ? It doesn't look like the one on the screenshot

I didn't, it just looks like that in game when it shows next to the overhead text

**Comment by cuackles** [comment]
> > Did you edit it ? It doesn't look like the one on the screenshot
> 
> I didn't, it just looks like that in game when it shows next to the overhead text 

Odd, gonna have to find out why it crunches them up so much and see if i can fix it

**Comment by MajManatee** [comment]
"ill look when i get home" is cuackles speak for "Ill try to maybe look into it in a week or so"

**Review by cuackles — APPROVED** [review]

**Comment by cuackles** [comment]
> "ill look when i get home" is cuackles speak for "Ill try to maybe look into it in a week or so"

Don't expose me like that

**Pull request merged**